sdk/src/main/scala/com/horizen/network/SyncStatusActor.scala

```diff
@@ -0,0 +1,275 @@
+          .asInstanceOf[Int]
+
+        // calculate the block correction needed to calculate the potential highest block
+        val blockCorrection = Await.result(sidechainNodeViewHolderRef ? GetDataFromCurrentView((view: View) =>
+          calculateBlockCorrection(view, currentHeightFromView)), timeoutDuration)
+          .asInstanceOf[Double]
+
+        // calculate the potential highest block
+        val currentTime: Long = timeProvider.time() / 1000
+        val timestampDifference = currentTime - sidechainBlock.timestamp
+        val blocksFromCurrentToTip = ((timestampDifference / params.consensusSecondsInSlot.toDouble) * blockCorrection).toInt
+
+        // set the current internal state
+        syncStatus = true
+        currentBlock = currentHeightFromView
+        startingBlock = currentHeightFromView
+        highestBlock = currentHeightFromView + blocksFromCurrentToTip
+
+        // broadcast sync start and its details, starting block is equal to current block in this case
+        val syncStatusMessage = new SyncStatus(true, BigInt(currentBlock), BigInt(startingBlock), BigInt(highestBlock))
+        context.system.eventStream.publish(NotifySyncStart(syncStatusMessage))
+
+      }
+      // if the node is already syncing we update the current block value adding one for every new block that is applied on the state
+      // we also check if the applied block timestamp is near the current timestamp, in this case we set the syncStatus to false and reset the state
+      else {
+        // update the current block counter
+        currentBlock += 1
+        if (checkAppliedBlockTimestamp(sidechainBlock.timestamp)) {
+          // reset the internal state
+          syncStatus = false; processNewBlocks = false
+          currentBlock = 0; startingBlock = 0; highestBlock = 0
+          // broadcast sync stop
+          context.system.eventStream.publish(NotifySyncStop())
+        }
+      }
+
+    }
+
+  }
+
+  // Calculate the block correction parameter retrieving the last n blocks (n = previousBlocksRange) and check how many slots were occupied
+  // if the previousBlocksRange value is less than the number of blocks present in the node history we just return the value 1
+  // this previous case can be encounter at node first startup without blocks in history
+  def calculateBlockCorrection(sidechainNodeView: View, currentBlockHeight: Int): Double = {
+    if (previousBlocksRange.compare(currentBlockHeight) < 0) {
+      val pastBlockInfo = Await.result(sidechainNodeViewHolderRef ? GetDataFromCurrentView((view: View) =>
+        retrieveBlockInfoByHeight(view, currentBlockHeight - previousBlocksRange)), timeoutDuration)
+        .asInstanceOf[SidechainBlockInfo]
+      val timestampDifference = timeProvider.time() - pastBlockInfo.timestamp
+      val expectedBlocksInRange = timestampDifference / params.consensusSecondsInSlot.toDouble
+      val slotCorrection = previousBlocksRange / expectedBlocksInRange
```

Comment:
`expectedBlocksInRange` can be less or equal than `previousBlocksRange`.
It means that the result of this method is `>= 1`
But the we use the result as a multiplicator:
`val blocksFromCurrentToTip = ((timestampDifference / params.consensusSecondsInSlot.toDouble) * blockCorrection).toInt`
why don't divide by?
`val blocksFromCurrentToTip = ((timestampDifference / params.consensusSecondsInSlot.toDouble) / blockCorrection).toInt` ?

Reply:
method and internal logic modified in the new class `SyncStatusUtil` but this comment was correct, multiply for a value `>= 1 `was not the correct way to retrieve the potential blocks from current to sidechain tip